WordPress/WordPressTest/BlogServiceRemoteXMLRPCTest.swift

```diff
@@ -0,0 +1,271 @@
+import XCTest
+@testable import WordPress
+
+class BlogServiceRemoteXMLRPCTest: XCTestCase {
+
+    let mockApi = MockWordPressOrgXMLRPCApi()
+    var blogServiceRemote: BlogServiceRemoteXMLRPC!
+
+    override func setUp() {
+        super.setUp()
+        blogServiceRemote = BlogServiceRemoteXMLRPC(api: mockApi, username: "", password: "")
+    }
+
+    func testCheckMultiAuthorRequest() {
+
+        let expectedParams = ["who" : "authors"]
+        blogServiceRemote.checkMultiAuthorWithSuccess(nil, failure: nil)
+        XCTAssertEqual(mockApi.methodPassedIn, "wp.getUsers")
+        XCTAssertTrue(checkParametersArray(mockApi.parametersPassedIn, expected: expectedParams),
+                      "\(expectedParams) not found in the parameter list")
+    }
+
+    func testIsMultiAuthor() {
+
+        var isMultiAuthor = false
+        let response = [["ID" : 2], ["ID" : 3]]
+        blogServiceRemote.checkMultiAuthorWithSuccess( {
+                isMultiAuthor = $0
+            }, failure: nil)
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertTrue(isMultiAuthor)
+    }
+
+    func testIsNotMultiAuthor() {
+
+        var isMultiAuthor = true
+        let response = [["ID" : 2]]
+        blogServiceRemote.checkMultiAuthorWithSuccess( {
+            isMultiAuthor = $0
+        }, failure: nil)
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertFalse(isMultiAuthor)
+    }
+
+    func testSyncPostTypesRequest() {
+
+        blogServiceRemote.syncPostTypesWithSuccess(nil, failure: nil)
+        XCTAssertEqual(mockApi.methodPassedIn, "wp.getPostTypes")
+        XCTAssertTrue(checkParametersArray(mockApi.parametersPassedIn, expected: nil),
+                      "Default parameters not found in the parameter list")
+    }
+
+    func testSingleSyncPostTypes() {
+
+        let name = "name"
+        let response = [1 : [name : name]]
+        var remotePost = [RemotePostType]()
+        blogServiceRemote.syncPostTypesWithSuccess({
+            remotePost = $0
+        }, failure: nil)
+        mockApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertEqual(remotePost.count, 1)
+        XCTAssertEqual(remotePost[0].name, name)
+    }
+
+    func testMultipleSyncPostTypes() {
+
+        let name = "name"
+        let response = ["1" : [name : name], "2" : [ name : name]]
```

Comment:
I'd rather use something that is (or looks like) a post type. When something fails, looking at `[1: ["name": "name"], 2: ["name": "name"]` can be hard to debug. My preferred approach for non-trivial responses is to add perform the request on a test blog, and save it to a file in the `Mock Data` folder (example response at https://gist.github.com/koke/b94d2cb7c655f7c24c4bfd56666ce4c2). Then tweak this file if you need to simulate some other data. `OHTTPStubs` really helps loading these mock responses, as you can see in `WordPressOrgXMLRPCApiTests`.